ledger/acctupdates.go

```diff
@@ -1478,14 +1481,14 @@ func (au *accountUpdates) lookupImpl(rnd basics.Round, addr basics.Address, with
 			withRewards = false
 		}
 
-		// Check if this is the most recent round, in which case, we can
-		// use a cache of the most recent account state.
-		if offset == uint64(len(au.deltas)) {
-			macct, ok := au.accounts[addr]
-			if ok {
+		// check if we've had this address modified in the past rounds. ( i.e. if it's in the deltas )
+		macct, indeltas := au.accounts[addr]
+		if indeltas {
+			// Check if this is the most recent round, in which case, we can
+			// use a cache of the most recent account state.
+			if offset == uint64(len(au.deltas)) {
 				return macct.data, nil
 			}
-		} else {
 			// Check if the account has been updated recently.  Traverse the deltas
 			// backwards to ensure that later updates take priority if present.
 			for offset > 0 {
```

Comment:
if ok { ...
					// the returned validThrough here is not optimal, but it still correct. We could get a more accurate value by scanning
					// the deltas forward, but this would be time consuming loop, which might not pay off.

Reply:
The comment is not relevant here, since `lookupWithRewardsImpl` doesn't return validity range. ( i.e. it can't due to the rewards )